### PR TITLE
Full JSON Output CLI

### PR DESCRIPTION
**Pull request by lawrencelomax**
- [x] Replace `Writer` on the `Runner` side with a higher-level representation, `EventReporter` that can serialize to a writer for an Event that occurs.
- [x] Add Event Names
- [x] Human Readable Reporter
- [x] JSON Reporter
- [x] Move JSON to `Format` rather than global configuration.

### CODE REVIEW

File: fbsimctl/FBSimulatorControlKit/Sources/EventReporter.swift [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        case .State:
+          return simulator.stateString
+        case .ProcessIdentifier:
+          guard let process = simulator.launchdSimProcess else {
+            return "no-process"
+          }
+          return process.processIdentifier.description
+        }
+      }
+      return tokens.joinWithSeparator(" ")
```
Comment (phyllipecesar) [inline fbsimctl/FBSimulatorControlKit/Sources/EventReporter.swift]: What does it mean? a name "a b" could be interpreted as two tokens? Should we use commas?

File: fbsimctl/FBSimulatorControlKit/Sources/EventReporter.swift [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    do {
+      self.writer.write(
+        try self.json.serializeToString([
+          "simulator" : self.simulator.jsonSerializableRepresentation(),
+          "event_name" : eventName.rawValue,
+          "event_type" : eventType.rawValue,
+          "subject" : subject.jsonSerializableRepresentation()
+        ])
+      )
+    } catch {
```
Comment (phyllipecesar) [inline fbsimctl/FBSimulatorControlKit/Sources/EventReporter.swift]: Sketchy =( Can we log anything?

File: fbsimctl/FBSimulatorControlKit/Sources/EventReporter.swift (hunk 1 above)
Comment (lawrencelomax) [inline fbsimctl/FBSimulatorControlKit/Sources/EventReporter.swift]: That's a good point, I think the simplest way for now is to quote tokens that have spaces in them in the same way that we separate argument tokens i.e. `'iPhone 5' list`

File: fbsimctl/FBSimulatorControlKit/Sources/EventReporter.swift (hunk 2 above)
Comment (lawrencelomax) [inline fbsimctl/FBSimulatorControlKit/Sources/EventReporter.swift]: I was thinking we `abort` instead, it's programmer error if an event is passed that is un-seralizable.

**Comment by lawrencelomax** [comment]
@facebook-github-bot shipit

**Pull request merged**